### ISSUE TITLE
Add support to split the specification file using $refs?

### ISSUE THREAD

**Issue by dod38fr** [cdimascio/express-openapi-validator#130]
Hi

Our swagger specification file is growing and we would like to [split it](https://azimi.me/2015/07/16/split-swagger-into-smaller-files.html).

OpenApi specification mentions:

> An OpenAPI document MAY be made up of a single document or be divided into multiple, connected parts at the discretion of the user. In the latter case, $ref fields MUST be used in the specification to reference those parts as follows from the JSON Schema definitions.

Unfortunately, I cannot find a way to use `$ref` with an URL to a local file. Something like:
```
        application/json:
          schema:
            $ref: 'test.yaml'
```
does not work, the file `test.yaml` is not loaded.

Is there any other way to split the specification file ?

All the best

**Comment by cdimascio** [comment]
you can split oas specs using the rules defined here: https://swagger.io/docs/specification/using-ref/ best!

**Comment by dod38fr** [comment]
yes, that's what I'm trying to do, but the `test.yaml` file is not loaded. This file is located in the same directory as the main swagger file.

**Comment by dod38fr** [comment]
Here's a patch to split one of your swagger test files. I would expect  `test/request.bodies.ref.spec.ts` to work even with this patch: https://paste.debian.net/1116984/

**Comment by cdimascio** [comment]
@dod38fr this is currently not supported. this looks like a good enhancement. i'll start digging into potential methods to resolve local files and add their conents to the schema.

**Comment by dod38fr** [comment]
After re-reading OpenApi spec, I believe that my patch above is slightly wrong. `test.yaml` should contain only the Schema object. In other word, it should not contain `openapi` field. See https://paste.debian.net/1117063 for an updated patch.

**Comment by cdimascio** [comment]
@dod38fr thanks. after digging into this issue and related works, i believe i can leverage json-schema-ref-parser to dereference any referenced schema files to construct the fully de-referenced schema. once we have a full de-referenced schema, express-openapi-validator should 'just work' as is.

**Comment by dod38fr** [comment]
Cool.. Thanks for following up so fast on this.

**Comment by cdimascio** [comment]
@dod38fr i've implemented this in `v2.17.0`, please give it a try. let me know if you run into issues.

**Comment by dod38fr** [comment]
It's working great :+1: 

Thanks a bunch !
